### ISSUE TITLE
version drop down broken

### ISSUE THREAD

**Issue by wkeese** [wkeese/old-api-viewer#8]
The new viewer doesn't support viewing the old API data, so using the version drop down should redirect to the old viewer for versions 1.3 --> 1.7.    I guess it can just set the location to http://dojotoolkit.org/api/VERSION/ etc.

Right now it just makes the tree disappear.

**Comment by kitsonk** [comment]
Looks like it is because there is no `tree.json` for that version.  Maybe the code could be change to validate the existance of that file before offering it up in the drop down list.

**Comment by wkeese** [comment]
But then the drop down list wouldn't have any listings at all, besides 1.8.

It's not just tree.json, the details.xml is different, and we no longer have files like provides.xml.    My point is that I don't want the new viewer code to support previous versions but this needs to be hidden from the user by navigating to the old viewer code.

**Comment by ttrenka** [comment]
Probably what you'll need to do is generate tree.json for the older versions of the API docs.  This should actually not be a problem; on deployments where generated data is cached (aka dojotoolkit.org), each version has a cached file for the tree; you can find it at `/data/[version]/cache/class-tree.json`.  From what I'm seeing in the dtk.org repo, versions 1.5 through 1.7 have this file.

You can probably just copy them and rename them for older versions.  Note that these json files are intended for use with ItemFileReadStore and not converted to the new dojo/store formats.

**Comment by ttrenka** [comment]
Links to the ones currently at dtk.org:

http://dojotoolkit.org/api/data/1.5/cache/class-tree.json
http://dojotoolkit.org/api/data/1.6/cache/class-tree.json
http://dojotoolkit.org/api/data/1.7/cache/class-tree.json

**Comment by wkeese** [comment]
Closed this, as it can be handled outside of the viewer.

**Comment by wkeese** [comment]
Note that I decided to handle displaying the old version data by using the old item.php and generate.php scripts.   Not sure if the old item.php is necessary or not, but the old generate.php definitely is, since the data format has changed quite a bit.

As for the tree, note that tree.json is a different format than class-tree.php outputs, since the latter was !ItemFileReadStore format.    But I wrote a conversion script in 0afa2c43693290e2d6969750d21a9ea12b2f8b56 and checked in the output to the website/api_data/VERSION directories.    It also required fixing class-tree.php to output proper JSON, in 3700653325340fadfc6b1b503bacea6095bf4a41.
